core/main/src/broker/thunder/user_data_migrator.rs

```diff
@@ -0,0 +1,918 @@
+// Copyright 2023 Comcast Cable Communications Management, LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+use std::{
+    collections::HashMap,
+    fmt,
+    fs::{File, OpenOptions},
+    path::Path,
+    sync::Arc,
+};
+
+use ripple_sdk::{
+    api::{device::device_peristence::StorageData, gateway::rpc_gateway_api::JsonRpcApiResponse},
+    log::{debug, error, info},
+    tokio::{
+        self,
+        net::TcpStream,
+        sync::{
+            mpsc::{self, Receiver, Sender},
+            Mutex,
+        },
+        time::{timeout, Duration},
+    },
+    utils::error::RippleError,
+};
+
+use serde::{Deserialize, Serialize};
+use serde_json::{json, Value};
+
+use crate::broker::endpoint_broker::{
+    self, BrokerCallback, BrokerOutput, BrokerRequest, EndpointBrokerState,
+};
+use crate::broker::rules_engine::{Rule, RuleTransformType};
+
+use futures::stream::SplitSink;
+use futures_util::SinkExt;
+
+use crate::broker::thunder_broker::ThunderBroker;
+use tokio_tungstenite::{tungstenite::Message, WebSocketStream};
+
+// TBD get the storage dir from manifest or other Ripple config file
+const RIPPLE_STORAGE_DIR: &str = "/opt/persistent/ripple";
+const USER_DATA_MIGRATION_CONFIG_FILE_NAME: &str = "user_data_migration_config.json";
+
+#[derive(Debug)]
+enum UserDataMigratorError {
+    ThunderRequestError(String),
+    ResponseError(String),
```

Comment:
Is `ResponseError` a thunder response error and if so should it follow the naming convention for request errors, e.g. `ThunderResponseError`?

Reply:
Will change this to `ThunderResponseError` for making it consistent.